modules/index/src/main/java/org/apache/ignite/internal/index/IndexManagementUtils.java

```diff
@@ -263,4 +265,31 @@ static boolean isPrimaryReplica(ReplicaMeta primaryReplicaMeta, ClusterNode loca
         return localNode.id().equals(primaryReplicaMeta.getLeaseholderId())
                 && timestamp.compareTo(primaryReplicaMeta.getExpirationTime()) < 0;
     }
+
+
+    /**
+     * Returns the catalog version in which the existing index created.
+     *
+     * <p>NOTE: Can be used outside of catalog/metastore events.</p>
+     *
+     * @param catalogService Catalog service.
+     * @param indexId Index ID of interest.
+     */
+    static int catalogVersionOfIndexCreation(CatalogService catalogService, int indexId) {
```

Comment:
Are we going to clean up old catalog versions somehow in the future? If yes, how is this method going to work?

Reply:
Yes, old versions of the catalog will be deleted. `AssertionError` will be thrown.
For now, I will only use this method to change the index status from `REGISTERED` to `BUILDING`.
Maybe you have a better suggestion?

I talked to the guys about storing the version of the catalog in which the index was created at `CatalogIndexDescriptor`, but they didn’t support me.

Reply:
> Maybe you have a better suggestion?

No, because I don't know how it is supposed to work.

> Yes, old versions of the catalog will be deleted. AssertionError will be thrown.

And how is this going to work? Or do you guarantee somehow that Catalog versions will not be removed when you are calling this method?

Reply:
Add todos